parquet-column/src/main/java/org/apache/parquet/column/Comparators.java

```diff
@@ -0,0 +1,273 @@
+      case DOUBLE:
+        return doubleComparator(logicalType);
+      case INT96:
+      case FIXED_LEN_BYTE_ARRAY:
+      case BINARY:
+        return binaryComparator(type, logicalType);
+      default:
+        throw new UnknownColumnTypeException(type);
+    }
+  }
+
+  public static BooleanComparator booleanComparator(OriginalType logicalType) {
+    if (logicalType != null)
+      throw new IllegalArgumentException("Invalid logical type for BOOLEAN: " + logicalType);
+    return BOOLEAN_COMPARATOR;
+  }
+
+  public static LongComparator int64Comparator(OriginalType logicalType) {
+    if (logicalType == null)
+      return LONG_COMPARATOR;
+    switch (logicalType) {
+      case INT_64:
+      case DECIMAL:
+      case TIME_MICROS:
+      case TIMESTAMP_MILLIS:
+      case TIMESTAMP_MICROS:
+        return LONG_COMPARATOR;
+      case UINT_64:
+        // TODO: return unsigned comparator
+        return LONG_COMPARATOR;
+      default:
+        throw new IllegalArgumentException("Invalid logical type for INT64: " + logicalType);
+    }
+  }
+
+  public static IntComparator int32Comparator(OriginalType logicalType) {
+    if (logicalType == null)
+      return INT_COMPARATOR;
+    switch (logicalType) {
+      case INT_8:
+      case INT_16:
+      case INT_32:
+      case DECIMAL:
+      case DATE:
+      case TIME_MILLIS:
+        return INT_COMPARATOR;
+      case UINT_8:
+      case UINT_16:
+      case UINT_32:
+        // TODO: return unsigned comparator
+        return INT_COMPARATOR;
+      default:
+        throw new IllegalArgumentException("Invalid logical type for INT32: " + logicalType);
+    }
+  }
+
+  public static FloatComparator floatComparator(OriginalType logicalType) {
+    if (logicalType != null)
+      throw new IllegalArgumentException("Invalid logical type for FLOAT: " + logicalType);
+    return FLOAT_COMPARATOR;
+  }
+
+  public static DoubleComparator doubleComparator(OriginalType logicalType) {
+    if (logicalType != null)
+      throw new IllegalArgumentException("Invalid logical type for DOUBLE: " + logicalType);
+    return DOUBLE_COMPARATOR;
+  }
+
+  public static BinaryComparator binaryComparator(PrimitiveType.PrimitiveTypeName type, OriginalType logicalType) {
+    switch (type) {
+      case INT96:
+        if (logicalType == null)
+          // TODO: what to return here?
+          return BINARY_COMPARATOR;
+        break;
+      case FIXED_LEN_BYTE_ARRAY:
+        if (logicalType == null)
+          // TODO: return lexicographical comparator
+          return BINARY_COMPARATOR;
+        switch (logicalType) {
+          case DECIMAL:
+            // TODO: return signed comparator
+            return BINARY_COMPARATOR;
+          case INTERVAL:
+            // TODO: return lexicographical comparator
+            return BINARY_COMPARATOR;
+        }
+        break;
+      case BINARY:
+        if (logicalType == null)
+          // TODO: return lexicographical comparator
```

Comment:
Here's a `Comparators` class from another project that has the unsigned byte-wise comparator for `ByteBuffer`, one for `CharSequence`, and some Java 8 utils for handling nulls.